indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -202,6 +205,16 @@ public CompactionTask(
     this.dimensionsSpec = dimensionsSpec == null ? dimensions : dimensionsSpec;
     this.metricsSpec = metricsSpec;
     this.segmentGranularity = segmentGranularity;
+    if (granularitySpec == null && segmentGranularity != null) {
+      this.granularitySpec = new UniformGranularitySpec(
+          segmentGranularity,
+          null,
+          null,
+          null
+      );
+    } else {
+      this.granularitySpec = granularitySpec;
```

Comment:
What happens if both segmentGranularity & granularitySpec are non-null?

Reply:
granularitySpec takes priority since segmentGranularity is deprecated.

Reply:
I'm not sure if we have a consistent stance on this but I think several places throw an exception if both the deprecated and new property are specified (such as `Checks.checkOneNotNullOrEmpty` usages).

I don't feel strongly about it, so I'm just mentioning this in case you feel that's a better approach.